src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentRequest.cs

```diff
@@ -28,49 +32,68 @@ internal class SilentRequest : RequestBase
         {
             _silentParameters = silentParameters;
 
-            _brokerStrategyLazy = new Lazy<ISilentAuthRequestStrategy>(() => brokerStrategyOverride ?? new SilentBrokerAuthStrategy(this,
+            _brokerStrategyLazy = new Lazy<ISilentAuthRequestStrategy>(() => brokerStrategyOverride ?? new BrokerSilentStrategy(this,
                                                                                                serviceBundle,
                                                                                                authenticationRequestParameters,
                                                                                                silentParameters,
                                                                                                serviceBundle.PlatformProxy.CreateBroker(null)));
-            _clientStrategy = clientStrategyOverride ?? new SilentClientAuthStretegy(this, serviceBundle, authenticationRequestParameters, silentParameters);
+            _clientStrategy = clientStrategyOverride ?? new CacheSilentStrategy(this, serviceBundle, authenticationRequestParameters, silentParameters);
 
             _logger = authenticationRequestParameters.RequestContext.Logger;
         }
 
-        internal async override Task PreRunAsync()
-        {
-            await _clientStrategy.PreRunAsync().ConfigureAwait(false);
-        }
-
         protected override async Task<AuthenticationResult> ExecuteAsync(CancellationToken cancellationToken)
         {
+            await UpdateRequestWithAccountAsync().ConfigureAwait(false);
+            bool isBrokerConfigured =
+              AuthenticationRequestParameters.IsBrokerConfigured &&
+              ServiceBundle.PlatformProxy.CanBrokerSupportSilentAuth();
+
             try
             {
-                _logger.Info("Attempting to acquire token using using local cache...");
-                return await _clientStrategy.ExecuteAsync(cancellationToken).ConfigureAwait(false);
+                if (AuthenticationRequestParameters.Account != null)
+                {
+                    _logger.Verbose("Attempting to acquire token using local cache...");
+                    return await _clientStrategy.ExecuteAsync(cancellationToken).ConfigureAwait(false);
+                }
+
+                if (!isBrokerConfigured)
+                {
+                    _logger.Verbose("No account passed to AcquireTokenSilent and no broker configured.  ");
+                    throw new MsalUiRequiredException(
+                       MsalError.UserNullError,
+                       MsalErrorMessage.MsalUiRequiredMessage,
+                       null,
+                       UiRequiredExceptionClassification.AcquireTokenSilentFailed);
+                }
+
+                _logger.Verbose("No account passed to AcquireTokenSilent. Some brokers may be able to log in user with a default account...");
```

Comment:
>Some brokers [](start = 74, length = 12)

looks like it's only WAM who can do this, so maybe just mention that so we don't get iOS/android devs asking about it.